pkg/controller/utils/utils.go

```diff
@@ -180,9 +182,6 @@ func WaitToAddTierWatch(tierName string, controller controller.Controller, c kub
 // will be generated.
 func AddNamespacedWatch(c controller.Controller, obj client.Object, metaMatches ...MetaMatch) error {
 	objMeta := obj.(metav1.ObjectMetaAccessor).GetObjectMeta()
-	if objMeta.GetNamespace() == "" {
```

Comment:
Pretty sure this was just over-aggressive validation - the code after this seems to support this being empty correctly.